File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int coins = 0;
    float dollars = -1;

    // Checks for valid values and converts to int value
    while (dollars <= 0)
    {
        dollars = get_float("Change owde: ");
    }
    int cents = round(dollars * 100);

    int remain = cents;
    while (remain >= 25)
    {
        remain -= 25;
        coins += 1;
    }
    while (remain >= 10)
    {
        remain -= 10;
        coins += 1;
    }
    while (remain >= 5)
    {
        remain -= 5;
        coins += 1;
    }
    while (remain >= 1)
    {
        remain -= 1;
        coins += 1;
    }

    printf("Minimum amount of coins: %i\n", coins);
}

File: pset6/cash/cash.py
#!/usr/bin/env python3
# -*- encoding: utf-8 -*-

import cs50
import math

counter = 0
money = 0

while money <= 0:
    money = round(cs50.get_float("Change owde: ") * 100)

while True:

    if money >= 25:
        money -= 25
        counter += 1
    elif money >= 10:
        money -= 10
        counter += 1
    elif money >= 5:
        money -= 5
        counter += 1
    elif money >= 1:
        money -= 1
        counter += 1
    else:
        break

print(f"Minimum amount of coins: {counter}\n")